networkit/cpp/graph/GraphTools.cpp

```diff
@@ -200,15 +200,66 @@ double density(const Graph &G) noexcept {
     return m / (n * (n - 1));
 }
 
-double volume(const Graph &G) {
-    if (G.isDirected())
-        throw std::runtime_error("Volume computation is only supported for undirected graphs.");
+double inVolume(const Graph &G, const std::unordered_set<node> &nodes) {
+    double inVolume = 0.0;
+    bool directed = G.isDirected() ? false : true;
+
+    // Compute volume of subgraph
+    if (!nodes.empty() && G.numberOfNodes() != nodes.size()) {
+        for (auto it = std::begin(nodes); it != std::end(nodes); ++it) {
+            inVolume += G.weightedDegreeIn(*it, directed);
```

Comment:
Counting self-loops twice for directed graphs seems wrong. The version for the whole graph counts them twice only for undirected graphs (as I would expect). I think this should be `!directed`. Or maybe not, as the variable is actually `true` if the graph is undirected?!